en/advanced-topics/cmake-learning.md

```diff
@@ -0,0 +1,350 @@
+
+        ```CMake
+        add_custom_command(TARGET <target>)
+                   PRE_BUILD | PRE_LINK | POST_BUILD
+                   COMMAND command1 [ARGS] [args1...]
+                   [COMMAND command2 [ARGS] [args2...] ...]
+                   [BYPRODUCTS [files...]]
+                   [WORKING_DIRECTORY dir]
+                   [COMMENT comment]
+                   [VERBATIM] [USES_TERMINAL]
+                   [COMMAND_EXPAND_LISTS])
+        ```
+
+        The meanings of some of the parameters are as follows:
+
+        - `TARGET`: specifies the target of the command run
+
+        - `COMMAND`: specifies the command line to be executed at build time.
+
+    For more information about the `add_custom_command` command, please refer to the official CMake documentation [add_custom_command](https://cmake.org/cmake/help/v3.16/command/add_custom_command.html?highlight=add_custom_command).
+
+### Linking library files
+
+The order of library files in ``target_link_libraries`` conforms to the rules of `gcc` linking order, i.e. the dependent library is placed after the library that depends on it.
+
+```CMake
+target_link_libraries(<target> [item1] [item2] [...]
+                      [[debug|optimized|general] <item>] ...)
+```
+
+`item` indicates a library file without a suffix name. By default, library dependencies are passed. When this target is linked to another target, the library linked to this target will also appear on the link line of the other target. For example:
+
+```CMake
+target_link_libraries(${LIB_NAME}
+    "-Wl,--whole-archive" cocos2d_jni "-Wl,--no-whole-archive"
+    cocos2d
+)
+```
+
+For more, see the official CMake documentation [target_link_libraries](https://cmake.org/cmake/help/latest/command/target_link_libraries.html).
+
+### Other commands
+
+- Print debug log messages
+
+    ```CMake
+    message(STATUS "my custom debug info")
+    ```
+
+- Manipulate file commands
+
+    ```CMake
+    file(_COPY "${COCOS_X_PATH}/cocos/bindings/manual/jsb_module_register.cpp"
+        DESTINATION ${CMAKE_CURRENT_LIST_DIR}/../common/Classes/)
+    ```
+
+    ``COPY`` means to copy files, directories and symbolic links to a target folder. The input path is the path relative to the current source code directory. The destination path is relative to the current build directory.
+
+- Loading and running CMake code from a file or module
+
+    ```CMake
+    include(${CMAKE_CURRENT_LIST_DIR}/../common/CMakeLists.txt)
+    ```
+
+## CMakeLists Usage Examples
+
+**Android.mk** is the Makefile of Android source code compilation system, which is used to compile dynamic libraries, static libraries, executables, apk, etc. of C++ in the system. The next step is to rewrite one `Android.mk` in [Tencent Cloud Multimedia Engine SDK](https://intl.cloud.tencent.com/document/product/607/15216) to ` CMakeLists.txt` for a simple example.
+
+The code in `Android.mk` is as follows:
+
+```
```

Comment:
```suggestion
```cmake
```

Reply:
Why?

Reply:
This part of code is not of cmake format, you don't need to add that

Reply:
I see that now... :-)

`*.mk` :-)